tasks/main.yml

```diff
@@ -1,83 +1,132 @@
 ---
-# NOTE: Anything that is in here is done every time you call the tripleo_role
+# NOTE: Anything that is in here is done every time you call the
+# tripleo_role
 # This means you should add variable parsing and setting here.
 
-- name: Set-up vars based on THT inputs
-  debug:
-    msg: "Pretending to parse vars; file: roles/tripleo_collectd/tasks/main.yml"
-
 - name: Report what deployment stage we're at
   debug:
     msg: "This is deployment stage {{ deploy_stage }}"
 
+- name: Create initial plugins list
+  set_fact:
+    collectd_plugins: "{{ (collectd_default_plugins + collectd_extra_plugins) | unique }}"
+
 - name: Set up conditionals
   set_fact:
     amqp_connection: "{{ (collectd_connection_type == 'amqp1')|default(false) }}"
     amqp_default_connection: "{{ (collectd_amqp_host == 'nil')|default(false) }}"
     amqp_default_interval: "{{ (collectd_amqp_interval|int == -666)|default(false) }}"
     collectd_connection: "{{ (collectd_connection_type == 'network')|default(false) }}"
-    gnocchi_connection:  "{{ (collectd_connection_type == 'gnocchi')|default(false) }}"
-    gnocchi_auth_basic:  "{{ (collectd_gnocchi_auth_mode == 'basic')|default(false) }}"
-    gnocchi_default_connection: "{{ ((collectd_gnocchi_server == 'nil') and (collectd_gnocchi_keystone_endpoint == 'nil'))|default(false) }}"
     sensubility_needs_sudo: "{{ (collectd_sensubility_exec_sudo_rule != '')| default(false) }}"
 
-# TODO: Add default for enable_stf, enable_sqlalchemy_collectd, enable_sensubility,
 - debug:
     var: "{{ item }}"
   with_items:
     - amqp_connection
     - amqp_default_connection
     - amqp_default_interval
     - collectd_connection
-    - gnocchi_connection
-    - gnocchi_auth_basic
-    - collectd_gnocchi_auth_mode
-    - gnocchi_default_connection
     - sensubility_needs_sudo
     - collectd_extra_plugins
 
-#TODO: set up this in default/main.yml, and then reset it to "" if enable_stf_plugin
 - name: "Update plugin list"
   set_fact:
-    collectd_extra_plugins: "{{ (collectd_extra_plugins + 'cpu df load connectivity intel_rdt ipmi procevent'.split() ) | unique }}"
+    collectd_plugins: "{{ (collectd_plugins + 'cpu df load connectivity intel_rdt ipmi procevent'.split() ) | unique }}"
```

Comment:
I couldn't find a copy of the THT's you're testing against, but if `collectd_plugins` is being passed in from Heat, I think you'll find that this code does not work. See here for details: https://gist.github.com/csibbitt/95db8af1ca882a9d7bee4ef09d53300d

Same for all of the other variables below. I'm going to be experimenting with this more during the week, but my current understanding is that if a variable is passed in using the method we've identified previously, you are unable to mutate them with set_fact.

Reply:
THT has the `CollectdDefaultPlugins` and `CollectdExtraPlugins` parameters [1], which translate to `collectd_default_plugins` and `collectd_extra_plugins`, respectively. 

collectd_config uses the `collectd_plugins` var, which is what's created in L10-13 in this file, by combining the parameters from THT (`collectd_default_plugins` and `collectd_extra_plugins`, as they're renamed in ansible) (as was previously done by THT in [2]).

As different plugins are enabled, the collectd_plugins var (created in this file) is extended, and is fed to collectd_config at the end.
[1] https://github.com/openstack/tripleo-heat-templates/blob/master/deployment/metrics/collectd-container-puppet.yaml#L79-L107
[2] https://github.com/openstack/tripleo-heat-templates/blob/master/deployment/metrics/collectd-container-puppet.yaml#L454-L472

Reply:
Okay, that looks like it will work. I see now where collectd_plugins comes from and how you've dealt with this problem. I'll try to emulate it.